### ISSUE TITLE
Bug: reprovision fails when switching PHP versions

### ISSUE THREAD

**Issue by robrecord** [roots/trellis#1394]
### Terms

- [X] I have read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
- [X] This request is not a duplicate of an existing issue
- [X] I have read the [docs](https://roots.io/docs/) and followed them (if applicable)
- [X] I have seached the [Roots Discourse](https://discourse.roots.io/) for answers and followed them (if applicable)
- [X] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

#### What's wrong?

When changing the PHP version in `group_vars/all/main.yml`, then reprovisioning, an error occurs:

```
TASK [php : Start php fpm service] *********************************************
System info:
  Ansible 2.13.1; Vagrant 2.2.18; Darwin
  Trellis version (per changelog): "Add built-in fail2ban filters"
---------------------------------------------------
Unable to start service php8.1-fpm: Job for php8.1-fpm.service failed because
the control process exited with error code.
See "systemctl status php8.1-fpm.service" and "journalctl -xe" for details.
fatal: [default]: FAILED! => {"changed": false}
```

#### What have you tried?

SSH into VM and running the above commands give this output:

```
sudo systemctl status php8.1-fpm.service
● php8.1-fpm.service - The PHP 8.1 FastCGI Process Manager
     Loaded: loaded (/lib/systemd/system/php8.1-fpm.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Thu 2022-06-23 03:46:46 PDT; 1min 39s ago
       Docs: man:php-fpm8.1(8)
    Process: 19475 ExecStart=/usr/sbin/php-fpm8.1 --nodaemonize --fpm-config /etc/php/8.1/fpm/php-fpm.conf (code=exited, status=78)
    Process: 19487 ExecStopPost=/usr/lib/php/php-fpm-socket-helper remove /run/php/php-fpm.sock /etc/php/8.1/fpm/pool.d/www.conf 81 (code=exited, status=0/SUCCESS)
   Main PID: 19475 (code=exited, status=78)

Jun 23 03:46:46 ... systemd[1]: Starting The PHP 8.1 FastCGI Process Manager...
Jun 23 03:46:46 ... php-fpm8.1[19475]: [23-Jun-2022 03:46:46] ERROR: Another FPM instance seems to already listen on /var/run/php-fpm-wordpress.sock
Jun 23 03:46:46 ... php-fpm8.1[19475]: [23-Jun-2022 03:46:46] ERROR: FPM initialization failed
Jun 23 03:46:46 ... systemd[1]: php8.1-fpm.service: Main process exited, code=exited, status=78/CONFIG
Jun 23 03:46:46 ... systemd[1]: php8.1-fpm.service: Failed with result 'exit-code'.
Jun 23 03:46:46 ... systemd[1]: Failed to start The PHP 8.1 FastCGI Process Manager.
```

#### What insights have you gained?

From the above output, we can see the old PHP instance is still running when we enable the new one. 

I also noticed the `roles/php/tasks/main.yml` script attempts to shut down the old processes, but does this after starting the new one. 

#### Possible solutions

As there is already provision for shutting down old processes, we should simply reverse the order so the old process is shut down before starting a new one. I have tested this and it works.

#### Temporary workarounds

It is also possible to manually shut down the old process after SSHing in. 

```sh
sudo systemctl stop php8.0-fpm.service # old process
# password is normally `vagrant`
```

### Steps To Reproduce

1. Spin up a new trellis instance
1. Change the PHP version
1. Reprovision

### Expected Behavior

New PHP version is provisioned

### Actual Behavior

Error on provisioning.

```
TASK [php : Start php fpm service] *********************************************
System info:
  Ansible 2.13.1; Vagrant 2.2.18; Darwin
  Trellis version (per changelog): "Add built-in fail2ban filters"
---------------------------------------------------
Unable to start service php8.1-fpm: Job for php8.1-fpm.service failed because
the control process exited with error code.
See "systemctl status php8.1-fpm.service" and "journalctl -xe" for details.
fatal: [default]: FAILED! => {"changed": false}
```

### Relevant Log Output

```shell
sudo systemctl status php8.1-fpm.service

● php8.1-fpm.service - The PHP 8.1 FastCGI Process Manager
     Loaded: loaded (/lib/systemd/system/php8.1-fpm.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Thu 2022-06-23 03:46:46 PDT; 1min 39s ago
       Docs: man:php-fpm8.1(8)
    Process: 19475 ExecStart=/usr/sbin/php-fpm8.1 --nodaemonize --fpm-config /etc/php/8.1/fpm/php-fpm.conf (code=exited, status=78)
    Process: 19487 ExecStopPost=/usr/lib/php/php-fpm-socket-helper remove /run/php/php-fpm.sock /etc/php/8.1/fpm/pool.d/www.conf 81 (code=exited, status=0/SUCCESS)
   Main PID: 19475 (code=exited, status=78)

Jun 23 03:46:46 ianbailie systemd[1]: Starting The PHP 8.1 FastCGI Process Manager...
Jun 23 03:46:46 ianbailie php-fpm8.1[19475]: [23-Jun-2022 03:46:46] ERROR: Another FPM instance seems to already listen on /var/run/php-fpm-wordpress.sock
Jun 23 03:46:46 ianbailie php-fpm8.1[19475]: [23-Jun-2022 03:46:46] ERROR: FPM initialization failed
Jun 23 03:46:46 ianbailie systemd[1]: php8.1-fpm.service: Main process exited, code=exited, status=78/CONFIG
Jun 23 03:46:46 ianbailie systemd[1]: php8.1-fpm.service: Failed with result 'exit-code'.
Jun 23 03:46:46 ianbailie systemd[1]: Failed to start The PHP 8.1 FastCGI Process Manager.
```

### Versions

HEAD at time of writing (1.14.0 + 3 commits). Vagrant 2.2.18. Parallels. macOS.
